### ISSUE TITLE
can't compile with node12

### ISSUE THREAD

**Issue by Kauto** [justadudewhohacks/native-node-utils#6]
When compiling face-recognition@0.9.4 on a  macOS Mojave Version 10.14.4 for Nodejs 12.1.0 I get a list of errors in native-node-utils

```
> face-recognition@0.9.4 install /Users/user/Desktop/Privat/code/face/node_modules/face-recognition
> node-gyp rebuild

  CXX(target) Release/obj.target/facerec/cc/index.o
In file included from ../cc/index.cc:2:
In file included from ../cc/utils.h:1:
In file included from /Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/Converters.h:1:
/Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/TypeConverters.h:25:28: error:
      too few arguments to function call, single argument 'context' was not
      specified
                return jsVal->Int32Value();
                       ~~~~~~~~~~~~~~~~~ ^
/Users/user/.node-gyp/12.1.0/include/node/v8.h:2569:3: note:
      'Int32Value' declared here
  V8_WARN_UNUSED_RESULT Maybe<int32_t> Int32Value(Local<Context> context) const;
  ^
/Users/user/.node-gyp/12.1.0/include/node/v8config.h:347:31: note:
      expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
In file included from ../cc/index.cc:2:
In file included from ../cc/utils.h:1:
In file included from /Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/Converters.h:1:
/Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/TypeConverters.h:43:29: error:
      too few arguments to function call, single argument 'context' was not
      specified
                return jsVal->Uint32Value();
                       ~~~~~~~~~~~~~~~~~~ ^
/Users/user/.node-gyp/12.1.0/include/node/v8.h:2567:3: note:
      'Uint32Value' declared here
  V8_WARN_UNUSED_RESULT Maybe<uint32_t> Uint32Value(
  ^
/Users/user/.node-gyp/12.1.0/include/node/v8config.h:347:31: note:
      expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
In file included from ../cc/index.cc:2:
In file included from ../cc/utils.h:1:
In file included from /Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/Converters.h:1:
/Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/TypeConverters.h:60:36: error:
      too few arguments to function call, single argument 'context' was not
      specified
                return (float)jsVal->NumberValue();
                              ~~~~~~~~~~~~~~~~~~ ^
/Users/user/.node-gyp/12.1.0/include/node/v8.h:2564:3: note:
      'NumberValue' declared here
  V8_WARN_UNUSED_RESULT Maybe<double> NumberValue(Local<Context> context) const;
  ^
/Users/user/.node-gyp/12.1.0/include/node/v8config.h:347:31: note:
      expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
In file included from ../cc/index.cc:2:
In file included from ../cc/utils.h:1:
In file included from /Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/Converters.h:1:
/Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/TypeConverters.h:78:17: error:
      no matching member function for call to 'BooleanValue'
                return jsVal->BooleanValue();
                       ~~~~~~~^~~~~~~~~~~~
/Users/user/.node-gyp/12.1.0/include/node/v8.h:2559:8: note: candidate
      function not viable: requires single argument 'isolate', but no arguments
      were provided
  bool BooleanValue(Isolate* isolate) const;
       ^
/Users/user/.node-gyp/12.1.0/include/node/v8.h:2562:51: note: candidate
      function not viable: requires single argument 'context', but no arguments
      were provided
                V8_WARN_UNUSED_RESULT Maybe<bool> BooleanValue(
                                                  ^
In file included from ../cc/index.cc:2:
In file included from ../cc/utils.h:1:
In file included from /Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/Converters.h:1:
/Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/TypeConverters.h:95:29: error:
      too few arguments to function call, single argument 'context' was not
      specified
                return jsVal->NumberValue();
                       ~~~~~~~~~~~~~~~~~~ ^
/Users/user/.node-gyp/12.1.0/include/node/v8.h:2564:3: note:
      'NumberValue' declared here
  V8_WARN_UNUSED_RESULT Maybe<double> NumberValue(Local<Context> context) const;
  ^
/Users/user/.node-gyp/12.1.0/include/node/v8config.h:347:31: note:
      expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
In file included from ../cc/index.cc:2:
In file included from ../cc/utils.h:1:
In file included from /Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/Converters.h:1:
/Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/TypeConverters.h:113:36: error:
      too few arguments to function call, single argument 'context' was not
      specified
                return (float)jsVal->NumberValue();
                              ~~~~~~~~~~~~~~~~~~ ^
/Users/user/.node-gyp/12.1.0/include/node/v8.h:2564:3: note:
      'NumberValue' declared here
  V8_WARN_UNUSED_RESULT Maybe<double> NumberValue(Local<Context> context) const;
  ^
/Users/user/.node-gyp/12.1.0/include/node/v8config.h:347:31: note:
      expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
In file included from ../cc/index.cc:2:
In file included from ../cc/utils.h:1:
In file included from /Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/Converters.h:1:
/Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/TypeConverters.h:131:46: error:
      no matching member function for call to 'ToString'
                return std::string(*Nan::Utf8String(jsVal->ToString()));
                                                    ~~~~~~~^~~~~~~~
/Users/user/.node-gyp/12.1.0/include/node/v8.h:2528:44: note: candidate
      function not viable: requires single argument 'context', but no arguments
      were provided
  V8_WARN_UNUSED_RESULT MaybeLocal<String> ToString(
                                           ^
/Users/user/.node-gyp/12.1.0/include/node/v8.h:2544:35: note: candidate
      function not viable: requires single argument 'isolate', but no arguments
      were provided
                    Local<String> ToString(Isolate* isolate) const);
                                  ^
In file included from ../cc/index.cc:2:
In file included from ../cc/utils.h:1:
In file included from /Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/Converters.h:2:
In file included from /Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/ArrayConverters.h:2:
/Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/InstanceConverter.h:18:48: error:
      no matching member function for call to 'ToObject'
                return Nan::ObjectWrap::Unwrap<Clazz>(jsVal->ToObject())...
                                                      ~~~~~~~^~~~~~~~
/Users/user/.node-gyp/12.1.0/include/node/v8.h:2532:44: note: candidate
      function not viable: requires single argument 'context', but no arguments
      were provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Object> ToObject(
                                           ^
/Users/user/.node-gyp/12.1.0/include/node/v8.h:2546:35: note: candidate
      function not viable: requires single argument 'isolate', but no arguments
      were provided
                    Local<Object> ToObject(Isolate* isolate) const);
                                  ^
In file included from ../cc/index.cc:2:
In file included from ../cc/utils.h:1:
In file included from /Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/Converters.h:2:
In file included from /Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/ArrayConverters.h:2:
/Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/InstanceConverter.h:22:48: error:
      no matching member function for call to 'ToObject'
                return Nan::ObjectWrap::Unwrap<Clazz>(jsVal->ToObject())...
                                                      ~~~~~~~^~~~~~~~
/Users/user/.node-gyp/12.1.0/include/node/v8.h:2532:44: note: candidate
      function not viable: requires single argument 'context', but no arguments
      were provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Object> ToObject(
                                           ^
/Users/user/.node-gyp/12.1.0/include/node/v8.h:2546:35: note: candidate
      function not viable: requires single argument 'isolate', but no arguments
      were provided
                    Local<Object> ToObject(Isolate* isolate) const);
                                  ^
In file included from ../cc/index.cc:2:
In file included from ../cc/utils.h:1:
In file included from /Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/Converters.h:2:
/Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/ArrayConverters.h:21:41: warning:
      'Get' is deprecated: Use maybe version [-Wdeprecated-declarations]
      if (!Converter::assertType(jsArr->Get(i))) {
                                        ^
/Users/user/.node-gyp/12.1.0/include/node/v8.h:3416:3: note: 'Get' has
      been explicitly marked deprecated here
  V8_DEPRECATE_SOON("Use maybe version", Local<Value> Get(uint32_t index));
  ^
/Users/user/.node-gyp/12.1.0/include/node/v8config.h:322:29: note:
      expanded from macro 'V8_DEPRECATE_SOON'
  declarator __attribute__((deprecated(message)))
                            ^
In file included from ../cc/index.cc:2:
In file included from ../cc/utils.h:1:
In file included from /Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/Converters.h:2:
/Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/ArrayConverters.h:33:54: warning:
      'Get' is deprecated: Use maybe version [-Wdeprecated-declarations]
  ...CastType obj = (CastType)Converter::unwrap(jsArr->Get(i));
                                                       ^
/Users/user/.node-gyp/12.1.0/include/node/v8.h:3416:3: note: 'Get' has
      been explicitly marked deprecated here
  V8_DEPRECATE_SOON("Use maybe version", Local<Value> Get(uint32_t index));
  ^
/Users/user/.node-gyp/12.1.0/include/node/v8config.h:322:29: note:
      expanded from macro 'V8_DEPRECATE_SOON'
  declarator __attribute__((deprecated(message)))
                            ^
In file included from ../cc/index.cc:2:
In file included from ../cc/utils.h:1:
In file included from /Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/Converters.h:2:
/Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/ArrayConverters.h:76:72: warning:
      'Get' is deprecated: Use maybe version [-Wdeprecated-declarations]
  ...(ArrayConverterType<Converter, T, CastType>::unwrap(&vec, jsArr->Get(i))) {
                                                                      ^
/Users/user/.node-gyp/12.1.0/include/node/v8.h:3416:3: note: 'Get' has
      been explicitly marked deprecated here
  V8_DEPRECATE_SOON("Use maybe version", Local<Value> Get(uint32_t index));
  ^
/Users/user/.node-gyp/12.1.0/include/node/v8config.h:322:29: note:
      expanded from macro 'V8_DEPRECATE_SOON'
  declarator __attribute__((deprecated(message)))
                            ^
In file included from ../cc/index.cc:2:
In file included from ../cc/utils.h:1:
In file included from /Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/Converters.h:2:
/Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/ArrayConverters.h:73:21: warning:
      comparison of integers of different signs: 'int' and 'uint32_t'
      (aka 'unsigned int') [-Wsign-compare]
                for (int i = 0; i < jsArr->Length(); i++) {
                                ~ ^ ~~~~~~~~~~~~~~~
/Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/ArrayConverters.h:87:21: warning:
      comparison of integers of different signs: 'int' and 'uint32_t'
      (aka 'unsigned int') [-Wsign-compare]
                for (int i = 0; i < jsArr->Length(); i++) {
                                ~ ^ ~~~~~~~~~~~~~~~
In file included from ../cc/index.cc:2:
In file included from ../cc/utils.h:1:
In file included from /Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/Converters.h:2:
In file included from /Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/ArrayConverters.h:2:
/Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/InstanceConverter.h:26:92: error:
      too few arguments to function call, single argument 'context' was not
      specified
  ...jsObj = Nan::NewInstance(Nan::New(Clazz::constructor)->GetFunction()).To...
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^
/Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/AbstractConverter.h:36:25: note:
      in instantiation of member function 'InstanceConverterType<Rect,
      dlib::rectangle>::wrap' requested here
                return ConverterType::wrap(val);
                                      ^
/Users/user/Desktop/Privat/code/face/node_modules/native-node-utils/src/AbstractConverter.h:60:25: note:
      in instantiation of member function
      'SingleTypeConverter<InstanceConverterType<Rect, dlib::rectangle>,
      dlib::rectangle>::wrap' requested here
                return ConverterType::wrap(val);
                                      ^
../cc/MmodRect.h:16:46: note: in instantiation of member function
      'AbstractConverter<SingleTypeConverter<InstanceConverterType<Rect,
      dlib::rectangle>, dlib::rectangle>, dlib::rectangle>::wrap' requested here
                info.GetReturnValue().Set(Rect::Converter::wrap(Converte...
                                                           ^
/Users/user/.node-gyp/12.1.0/include/node/v8.h:5947:3: note:
      'GetFunction' declared here
  V8_WARN_UNUSED_RESULT MaybeLocal<Function> GetFunction(
  ^
/Users/user/.node-gyp/12.1.0/include/node/v8config.h:347:31: note:
      expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
5 warnings and 10 errors generated.
```

**Comment by goulash1971** [comment]
See https://github.com/justadudewhohacks/opencv4nodejs/issues/538

At first glance looks like it's a bunch of NAN changes that rolled-out after `node 10` - took some time to dive into the code and fixed `FF_NEW_INSTANCE` in `macros.h` ...

```
// #define FF_NEW_INSTANCE(ctor) Nan::NewInstance(Nan::New(ctor)->GetFunction()).ToLocalChecked()
#define FF_NEW_INSTANCE(ctor) Nan::NewInstance(Nan::GetFunction(Nan::New(ctor)).ToLocalChecked()).ToLocalChecked()
```

... but bunch of others around the place.

**Comment by justadudewhohacks** [comment]
Thanks for reporting, fix coming soon...

**Comment by justadudewhohacks** [comment]
Fixed in native-node-utils 0.1.6 and opencv4nodejs 4.16.0.
